cloud/amazon/ec2_vpc_igw_facts.py

```diff
@@ -0,0 +1,161 @@
+    required: false
+    default: null
+  InternetGatewayIds:
+    description:
+      - Get details of specific Internet Gateway ID
+      - Provide this value as a list
+    required: false
+    default: None
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather facts about all Internet Gateways for an account or profile
+  ec2_vpc_igw_facts:
+    region: ap-southeast-2
+    profile: production
+  register: igw_facts
+
+- name: Gather facts about a filtered list of Internet Gateways
+  ec2_vpc_igw_facts:
+    region: ap-southeast-2
+    profile: production
+    filters:
+        "tag:Name": "igw-123"
+  register: igw_facts
+
+- name: Gather facts about a specific internet gateway by InternetGatewayId
+  ec2_vpc_igw_facts:
+    region: ap-southeast-2
+    profile: production
+    InternetGatewayIds: igw-c123f6a7 
+  register: igw_facts
+'''
+
+RETURN = '''
+internet_gateways:
+    description: The internet gateways for the account
+    returned: always
+    type: list
+
+changed:
+    description: True if listing the internet gateways succeeds
+    type: bool
+    returned: always
+'''
+
+try:
+    import json
+    import botocore
+    import boto3   
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_internet_gateway_info(internet_gateway):
+    internet_gateway_info = {'InternetGatewayId': internet_gateway['InternetGatewayId'],
+                             'Attachments': internet_gateway['Attachments'],
+                             'Tags': internet_gateway['Tags']
+               }
+    return internet_gateway_info
+
+
+def list_internet_gateways(client, module):
+    dryrun = module.params.get("DryRun")
+    all_internet_gateways_array = []
+    params = dict()
+
+    if module.params.get('filters'):
+        params['Filters'] = []
+        for key, value in module.params.get('filters').iteritems():
+            temp_dict = dict()
+            temp_dict['Name'] = key
+            if isinstance(value, basestring):
+                temp_dict['Values'] = [value]
+            else:
+                temp_dict['Values'] = value
+            params['Filters'].append(temp_dict)
+
+    if module.params.get("DryRun"):
+        params['DryRun'] = module.params.get("DryRun")
+
+    if module.params.get("InternetGatewayIds"):
+        params['InternetGatewayIds'] = module.params.get("InternetGatewayIds")
+
+    try:
+        all_internet_gateways = client.describe_internet_gateways(**params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    for internet_gateway in all_internet_gateways['InternetGateways']:
+        all_internet_gateways_array.append(get_internet_gateway_info(internet_gateway))
+
+    module.exit_json(internet_gateways=all_internet_gateways_array)
```

Comment:
You need to change the boto result to be ansible friendly by changing from CamelCase to snake_case.

Use the helper function 'camel_dict_to_snake_dict'.  See GUIDELINES and ec2_snapshot_facts for example